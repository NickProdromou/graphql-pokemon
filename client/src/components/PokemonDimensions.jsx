import React from "react";
import PropTypes from "prop-types";
import Styled from "styled-components";

import { type } from "../style/mixins/index";
import { spacing, colours } from "../style/variables";

const PokemonDimensions = ({ min, max }) => (
  <StyledDimensions>
    <div>
      <span className="DimensionTitle">min:</span>
      <span className="DimensionAmount">{min}</span>
    </div>
    <div>
      <span className="DimensionTitle">max:</span>
      <span className="DimensionAmount">{max}</span>
    </div>
  </StyledDimensions>
);

PokemonDimensions.propTypes = {
  min: PropTypes.string.isRequired,
  max: PropTypes.string.isRequired
};

const StyledDimensions = Styled.div`
  padding: ${spacing.small.level2};
  background: ${colours.borderColor};
  display: flex;
  flex-direction: column;

  .DimensionTitle {
    ${type("ui")}
    margin-top: 0;
    margin-right: ${spacing.small.level2};
  }

  .DimensionAmount {
    ${type("ui")}
    font-weight: bold;
  }
`;

export default PokemonDimensions;
