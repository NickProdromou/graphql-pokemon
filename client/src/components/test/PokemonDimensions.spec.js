import React from "react";
import renderer from "react-test-renderer";
import PokemonDimensions from "../PokemonDimensions";

describe("PokemonDimensions component", () => {
  const props = {
    type: "Fire"
  };

  test("renders", () => {
    const props = {
      min: "34",
      max: "65",
    };

    const component = renderer.create(
      <PokemonDimensions
        min={props.min}
        max={props.max}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
