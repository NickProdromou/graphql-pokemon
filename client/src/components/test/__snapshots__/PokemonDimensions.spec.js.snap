// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDimensions component renders 1`] = `
<div
  className="sc-bdVaJa cKBsET"
>
  <div>
    <span
      className="DimensionTitle"
    >
      min:
    </span>
    <span
      className="DimensionAmount"
    >
      34
    </span>
  </div>
  <div>
    <span
      className="DimensionTitle"
    >
      max:
    </span>
    <span
      className="DimensionAmount"
    >
      65
    </span>
  </div>
</div>
`;
